src/main/kotlin/uk/gov/justice/digital/hmpps/calculatereleasedatesapi/service/timeline/BookingTimelineService.kt

```diff
@@ -209,6 +224,153 @@ class BookingTimelineService(
+
+      if (numberOfDaysToAdjust != 0L) {
+        return applyAdjustment(
+          calculation,
+          releaseDateType,
+          commencementDate,
+          rule,
+          numberOfDaysToAdjust,
+        )
+      }
+    }
+
+    return calculation
+  }
+
+  private fun applyAdjustment(
+    calculation: CalculationResult,
+    releaseDateType: ReleaseDateType,
+    commencementDate: LocalDate,
+    rule: CalculationRule,
+    numberOfDaysToAdjust: Long,
+  ): CalculationResult {
+    var updatedCalculation = calculation.copy(
+      dates = calculation.dates.plus(
+        releaseDateType to calculation.dates[releaseDateType]!!.plusDays(numberOfDaysToAdjust),
+      ),
+    )
+    log.info("$releaseDateType after adjustment: ${updatedCalculation.dates[releaseDateType]}")
+
+    if (updatedCalculation.dates[releaseDateType]?.isBefore(commencementDate) == true &&
+      rule in listOf(
+        CalculationRule.SDS_EARLY_RELEASE_ADJUSTED_TO_TRANCHE_1_COMMENCEMENT,
+        CalculationRule.SDS_EARLY_RELEASE_ADJUSTED_TO_TRANCHE_2_COMMENCEMENT,
+      )
+    ) {
+      updatedCalculation = updatedCalculation.copy(
+        dates = updatedCalculation.dates.plus(
+          releaseDateType to commencementDate,
+        ),
+      )
+    }
+
+    return updateBreakdownWithTrancheAdjustedRule(updatedCalculation, releaseDateType)
+  }
+
+  private fun calculateAdjustmentDaysAfterCommencement(
+    adjustments: Adjustments,
+    commencementDate: LocalDate,
+    crd: LocalDate,
+  ): Long {
+    val ualDays = adjustments.getOrEmptyList(UNLAWFULLY_AT_LARGE)
+      .filter { it.appliesToSentencesFrom.isAfterOrEqualTo(commencementDate) && it.appliesToSentencesFrom.isBefore(crd) }
+      .sumOf { it.numberOfDays }
+
+    val adaDays = adjustments.getOrEmptyList(ADDITIONAL_DAYS_AWARDED)
+      .filter { it.appliesToSentencesFrom.isAfterOrEqualTo(commencementDate) && it.appliesToSentencesFrom.isBefore(crd) }
+      .sumOf { it.numberOfDays }
+
+    val radaDays = adjustments.getOrEmptyList(RESTORATION_OF_ADDITIONAL_DAYS_AWARDED)
+      .filter { it.appliesToSentencesFrom.isAfterOrEqualTo(commencementDate) && it.appliesToSentencesFrom.isBefore(crd) }
+      .sumOf { it.numberOfDays }
+
+    val totalAdjustmentsToApply = ualDays + adaDays - radaDays
+    log.info("Total adjustments to apply $totalAdjustmentsToApply for commencement date $commencementDate (UAL $ualDays + ADAs $adaDays + RADA $radaDays)")
+
+    return totalAdjustmentsToApply.toLong()
+  }
+
+  private fun updateBreakdownWithTrancheAdjustedRule(
+    calculation: CalculationResult,
+    releaseDateType: ReleaseDateType,
+  ): CalculationResult {
+    val existingBreakdown = calculation.breakdownByReleaseDateType[releaseDateType]
+    return calculation.copy(
+      breakdownByReleaseDateType = calculation.breakdownByReleaseDateType.plus(
+        releaseDateType to existingBreakdown!!.copy(
+          rules = existingBreakdown.rules.plus(ADJUSTED_AFTER_TRANCHE_COMMENCEMENT),
```

Comment:
Could this be added as a `rulesWithExtraAdjustments` so we can see the number of days that the release date was adjusted by?

Reply:
yes make sense - done

In the next ticket (where this rule will be used for the hints) will add some unit tests to test the braekdown explicitly too